processor/processor.go

```diff
@@ -460,7 +461,8 @@ var (
 
 func loadConfig() {
 	config.RegisterBoolConfigVariable(true, &enablePipelining, false, "Processor.enablePipelining")
-	config.RegisterIntConfigVariable(1, &pipelineBufferedItems, false, 1, "Processor.pipelineBufferedItems")
+	config.RegisterIntConfigVariable(0, &pipelineBufferedItems, false, 1, "Processor.pipelineBufferedItems")
+	config.RegisterIntConfigVariable(5, &subJobCount, false, 1, "Processor.subJobCount")
```

Comment:
@saurav-malani  when using a very low `maxEventsToProcess`, e.g. 100, this would split events in 5 subJobs of 20 items each. Similarly, when using a very high  `maxEventsToProcess`, e.g. 100000, this would split events in 5 subJobs of 20000 items each. Based on the above and given the fact that you have found 2000 items per batch to be the most efficient number, should we use a configurable `subJobSize` (default 2000) and derive the `subJobCount` from it, instead of the opposite?

Reply:
makes sense. I assumed that `maxEventsToProcess` is 10k always. But, since it's configurable, this is not a correct assumption. Thanks for pointing it out.

Reply:
here is some sample code for splitting

```
type subJob struct {
	jobs  []*jobsdb.JobT
	index int
	size  int
}

func splitJobs(jobs []*jobsdb.JobT, batchSize int) []subJob {
	res := []subJob{}
	jobsSize := len(jobs)
	if batchSize >= jobsSize {
		res = append(res, subJob{jobs, 0, 1})
		return res
	}
	subJobCount := jobsSize / batchSize
	if jobsSize%batchSize > 0 {
		subJobCount++
	}
	for i := 0; i < subJobCount; i++ {
		startIdx := i * batchSize
		endIdx := (i + 1) * batchSize
		if endIdx > jobsSize {
			endIdx = jobsSize
		}
		res = append(res, subJob{jobs[startIdx:endIdx], 0, 1})
	}
	return res
}
```

and the relevant test for possible scenarios

```
type subJob struct {
	jobs  []*jobsdb.JobT
	index int
	size  int
}

func splitJobs(jobs []*jobsdb.JobT, batchSize int) []subJob {
	res := []subJob{}
	jobsSize := len(jobs)
	if batchSize >= jobsSize {
		res = append(res, subJob{jobs, 0, 1})
		return res
	}
	subJobCount := jobsSize / batchSize
	if jobsSize%batchSize > 0 {
		subJobCount++
	}
	for i := 0; i < subJobCount; i++ {
		startIdx := i * batchSize
		endIdx := (i + 1) * batchSize
		if endIdx > jobsSize {
			endIdx = jobsSize
		}
		res = append(res, subJob{jobs[startIdx:endIdx], 0, 1})
	}
	return res
}
```

Reply:
I have created a `splitter` func to convert jobs into subJobs.